src/adapters/underdogmedia.js

```diff
@@ -5,71 +5,108 @@ var utils = require('../utils.js');
 
 var UnderdogMediaAdapter = function UnderdogMediaAdapter() {
 
-  var getJsStaticUrl = window.location.protocol + '//bid.underdog.media/udm_header_lib.js';
+  const UDM_ADAPTER_VERSION = "1.0.0";
+  var getJsStaticUrl = window.location.protocol + '//udmserve.net/udm/img.fetch?tid=1;dt=9;callback=$$PREBID_GLOBAL$$.handleUnderdogMediaCB;';
+  var bidParams = {};
 
   function _callBids(params) {
-    if (typeof window.udm_header_lib === 'undefined') {
-      adloader.loadScript(getJsStaticUrl, function () { bid(params); });
-    } else {
-      bid(params);
-    }
-  }
+    bidParams = params;
+    var sizes = [];
+    var siteId = 0;
 
-  function bid(params) {
-    var bids = params.bids;
-    var mapped_bids = [];
-    for (var i = 0; i < bids.length; i++) {
-      var bidRequest = bids[i];
-      var callback = bidResponseCallback(bidRequest);
-      mapped_bids.push({
-        sizes: bidRequest.sizes,
-        siteId: bidRequest.params.siteId,
-        bidfloor: bidRequest.params.bidfloor,
-        placementCode: bidRequest.placementCode,
-        divId: bidRequest.params.divId,
-        subId: bidRequest.params.subId,
-        callback: callback
-      });
+    for (var bidParam of bidParams.bids) {
+      sizes = utils.flatten(sizes,utils.parseSizesInput(bidParam.sizes));
+      siteId = bidParam.params.siteId;
     }
-    var udmBidRequest = new window.udm_header_lib.BidRequestArray(mapped_bids);
-    udmBidRequest.send();
+    adloader.loadScript(getJsStaticUrl + "sid=" + siteId + ";sizes=" + sizes.join(","), null, false);
   }
 
-  function bidResponseCallback(bid) {
-    return function (bidResponse) {
-      bidResponseAvailable(bid, bidResponse);
-    };
-  }
+  function _callback(response) {
+
+    var mids = response.mids;
+    for (var bidParam of bidParams.bids) {
+
+      var filled = false;
+      for (var mid of mids) {
+
+        if (mid.useCount > 0) {
+          continue;
+        }
+        if (!mid.useCount) {
+          mid.useCount = 0;
+        }
+        var size_not_found = true;
+        for (var size of bidParam.sizes) {
```

Comment:
this throws an uncaught exception if sizes are defined as a single array. Example:  `adUnit.sizes=[300,250]` instead of `adUnit.sizes=[[300,250]]`